src/convertToObject.js

```diff
@@ -6,7 +6,67 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  let formatData = sourceString.split(';').map((item) => item.split(':'));
+
+  formatData = formatData.map((item) => {
+    for (let i = 0; i < item.length; i++) {
+      if (item[i].includes('\n')) {
+        do {
+          item[i] = item[i].replace('\n', '');
+        } while (item[i].includes('\n'));
```

Comment:
The loop for replacing newline characters can be optimized. Instead of using a `do...while` loop, consider using `item[i] = item[i].replace(/\n/g, '');` to replace all newline characters in one go.